control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -707,6 +707,14 @@ func (r *HostedControlPlaneReconciler) reconcile(ctx context.Context, hostedCont
 		if err := r.reconcileManagedEtcd(ctx, hostedControlPlane, releaseImage, createOrUpdate); err != nil {
 			return fmt.Errorf("failed to reconcile etcd: %w", err)
 		}
+		ready, err := util.IsStatefulSetReady(ctx, r, manifests.EtcdStatefulSet(hostedControlPlane.Namespace))
```

Comment:
should this rather check for the `EtcdAvailable` condition which contains more knowledge e.g. quorum beyond the sts being ready https://github.com/openshift/hypershift/blob/405cd32e1045d7487534c99b28c6e245c34f22e7/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L3076